samza-core/src/main/scala/org/apache/samza/diagnostics/DiagnosticsManager.java

```diff
@@ -141,36 +171,83 @@ public void addExceptionEvent(DiagnosticsExceptionEvent diagnosticsExceptionEven
     this.exceptions.add(diagnosticsExceptionEvent);
   }
 
+  public void addProcessorStopEvent(String processorId, String resourceId, String host, int exitStatus) {
+    this.containerStops.add(new ProcessorStopEvent(processorId, resourceId, host, exitStatus));
+    LOG.info("Added stop event for Container Id: {}, resource Id: {}, host: {}, exitStatus: {}", processorId,
+        resourceId, host, exitStatus);
+  }
+
   private class DiagnosticsStreamPublisher implements Runnable {
 
     @Override
     public void run() {
+
+      Map<String, Map<String, Object>> metricsMessage = new HashMap<>();
+
+      // Publish job-related params (if not already published)
+      if (!jobParamsEmitted) {
+        metricsMessage.putIfAbsent(GROUP_NAME_FOR_DIAGNOSTICS_MANAGER, new HashMap<>());
+        metricsMessage.get(GROUP_NAME_FOR_DIAGNOSTICS_MANAGER).put(CONTAINER_COUNT_METRIC_NAME, containerCount);
```

Comment:
yes, since messages from container and AM can arrive in any order
publishing from both made the read logic simple
Since its only published once per container, i didn't think there was significant overhead.